TitechAppLite/Decoder/IcsDecoder.swift

```diff
@@ -0,0 +1,119 @@
+                    let arr = line.components(separatedBy: ":")
+                    if arr.count < 2 {
+                        return
+                    }
+                    tmpdtstamp = ISO8601StringtoDateTime(dateString: arr[1], tzoffset: tzoffset, fmt: fmt)
+                }
+                if line.hasPrefix(DTSTART) {
+                    let arr = line.components(separatedBy: ":")
+                    if arr.count < 2 {
+                        return
+                    }
+                    tmpdtstart = ISO8601StringtoDateTime(dateString: arr[1], tzoffset: tzoffset, fmt: fmt)
+                }
+                if line.hasPrefix(DTEND) {
+                    let arr = line.components(separatedBy: ":")
+                    if arr.count < 2 {
+                        return
+                    }
+                    tmpdtend = ISO8601StringtoDateTime(dateString: arr[1], tzoffset: tzoffset, fmt: fmt)
+                }
+                if line.hasPrefix(LOCATION) {
+                    let l = line.replacingOccurrences(of: LOCATION + ":", with: "")
+                    tmplocation = l
+                }
+                if line.hasPrefix(DESCRIPTION) {
+                    let l = line.replacingOccurrences(of: DESCRIPTION + ":", with: "")
+                    tmpdescription = l
+                }
+                if line.hasPrefix(SUMMARY) {
+                    let l = line.replacingOccurrences(of: SUMMARY + ":", with: "")
+                    tmpsummary = l
+                }
+                if line.hasPrefix(UID) {
+                    let l = line.replacingOccurrences(of: UID + ":", with: "")
+                    tmpuid = l
+                }
+                if line.hasPrefix(END) {
+                    guard
+                        let dtstamp = tmpdtstamp,
+                        let dtstart = tmpdtstart,
+                        let dtend = tmpdtend,
+                        let uid = tmpuid
+                        else {
+                            return
+                    }
+                    result.append(
+                        ICSEvent(
+                            dtstamp: dtstamp,
+                            dtstart: dtstart,
+                            dtend: dtend,
+                            location: tmplocation ?? "",
+                            description: tmpdescription ?? "",
+                            summary: tmpsummary ?? "",
+                            id: uid
+                        )
+                    )
+                    flag = false
+                    
+                }
+            }
+            
+        }
+        return result
+    }
+
+    static func ISO8601StringtoDateTime(dateString: String, tzoffset: String, fmt: ISO8601DateFormatter) -> Date? {
+        var opt: ISO8601DateFormatter.Options = [.withFullDate, .withFullTime]
+        opt.remove([.withDashSeparatorInDate, .withColonSeparatorInTime, .withColonSeparatorInTimeZone])
+        fmt.formatOptions = opt
```

Comment:
> Important
Resetting this property can incur a significant performance cost, as it may cause internal state to be regenerated.

https://developer.apple.com/documentation/foundation/iso8601dateformatter/1643324-formatoptions

ここを呼んでいただきたいです 🙇

Reply:
static var にして、optionを最初意外に変えないようにしました。